Shell/Options.cpp

```diff
@@ -1502,16 +1502,14 @@ void Options::init()
     _condensation.tag(OptionTag::INFERENCES);
     _condensation.onlyUsefulWith(ProperSaturationAlgorithm());
 
-    _demodulationRedundancyCheck = ChoiceOptionValue<DemodulationRedunancyCheck>("demodulation_redundancy_check","drc",DemodulationRedunancyCheck::ON,{"off","encompass","on"});
+    _demodulationRedundancyCheck = BoolOptionValue("demodulation_redundancy_check","drc",true);
```

Comment:
Do we know whether becoming incomplete is ever a good idea, strategy-scheduling wise?

Reply:
It is definitely a good idea (to keep drc=false). That's why Andrei was not so impressed by this when we explained encompass it in Vienna.

Experimentally: if we were comparing "old with drc=on:" Union "old with drc=off" this would most likely beat "new drc-on" (which does the encompass). @mezpusz, if you have energy, such experiment could be revealing. (Just for Discount is enough.)

Reply:
Do you mean `drc=off` would beat encompassment? Sure, I can run on more experiment. 

Btw, I never understood why demodulation on negative literals was sometimes disallowed with the old `drc=on`. Wouldn't an `s != t` be always bigger than `s = r` where `s` is the maximal term of the three, regardless of how `t` and `r` compare? Maybe this is why `drc=off` was preferred over `drc=on`? We used `drc=off` in induction schedules a lot because otherwise we couldn't rewrite the maximal term in a ground disequation in many cases.

Reply:
Using Discount, the uniques changed as follows:
| `drc` | `on` | `off` |
|--------|--------|--------|
| old | 91 | 244 |
| new | 44 | 85 |
which means that encompassment makes the gap between `off` and `on` a bit tighter, but there is still a gap.

Reply:
> Completeness:

I admit I don't remember the theory by heart, but there is again a comment (that's now getting deleted) about it. The problematic case seems to be:

s = t into s != t1 \\/ C

with s > t (we know always) plus t > t1 plus {s = t} > C. I guess with t > t1 we have {s = t} > {s != t1} (the polarity only comes in last). And standard redundancy asks that the reducer be smaller than the reduced...

> Maybe this is why drc=off was preferred over drc=on?

I think it's simpler. Rewriting things is good. The more (with drc=off) the merrier. (Also its more efficient not to check exotic, corner-case conditions.) Compromising completeness can hurt, though. At least sometimes, if only rarely. (That's why we had drc=on.)

Thus: Why not be too excited about encompassment? Because most of the problems that encompass solves and old-drc=on does not, will be covered by drc=off.

> Experiment:

Could you please reveal plain number of solved for every row at least for one of on/off?

Reply:
> there is again a comment (that's now getting deleted) about it.

I mean that that comment (and the corresponding implementation) was maybe a bit too restrictive. Of course it depends on how we define > between clauses and literals.

> Rewriting things is good. The more (with drc=off), the merrier.

This also makes sense, and it's hard to tell when a particular equality needed for completeness is used at all.

Anyways, here are the number of solved problems:
Otter off: 9813
Otter old on: 9562
Otter new on (encompass): 9776

Discount off: 9992
Discount old on: 9840
Discount new on (encompass): 9953

Reply:
This is an interesting point of view which I haven't considered yet. You're right, encompassment+off gives less variation then the other pair. I fail to see why there are so many uniques of "old on" compared to encompassment, but the way it skips demodulations that both `off` and `encompass` allow gives more variation. One more question is whether we care about completeness or not.

But the bottomline is, I am not so sure about this PR anymore.

Reply:
My part: I have around 1300 random strategies evaluated (with varying instr limits) over 9090 FOF problems from the latest TPTP. The drc values were sampled with the following frequencies: encompass:500,on:500,off:354. (The strategies in the set, however, are not independent samples, as there was pressure to solve unsolved problems or at least improve solutions times.) Now these strategies cover 7210 of 9090 (this includes sat results, but should be negligible). If we 

1) drop the drc=off strategies, we lose
41 solvable problems, for 1503 we will only get a worse best solution time, accumulated instr_lost: 2005432 of 7833518 
2) drop the drc=encompass strategies, we lose
54 solvable problems, for 1816 we will only get a worse best solution time, accumulated instr_lost: 2223449 of 7833518 
3) drop the default drc=on (which is a bit tricky to filter, because defaults are not mentioned in the strat string, and for strategies with fd=bd=off the default does not meaningfully count), we lose
42 of solvable problems, for 1497 we will only get a worse best solution time, accumulated instr_lost 2966861 of 7833518
 
I think the conclusion is, from the point of view of strategy scheduling, that we want to keep all three option values.

Still, I like to code simplification too :)

Reply:
Honestly, I don't understand much of the data but I believe you that we need all of them.:) So, my proposal is that we keep all three of them, maybe switch the default to `encompass`, or just leave this as it is and come back to it after the Vampire developer meeting (hope you're coming too!).

Reply:
I agree we should keep this open for now. It serves as a good example of a tricky decision we often face during vampire development and leaving is as an item for a developer meeting agenda could be useful.